### PR TITLE
Examples Update + Code Refactor

### PR DESCRIPTION
**Pull request by cyrus2281**
## What is the purpose of this change?
Refactoring the code base to ensure all examples work as expected + adding more examples

## How is this accomplished?
Refactoring in some areas of code including storage manager, langchain_vector_store_embedding_base, and example files.

## Anything reviews should focus on/be aware of?
Check summary for list of changes

## PR Summary

- [Removed StorageManager](https://github.com/SolaceDev/solace-ai-connector/commit/04c7c115cd31840d4a91c4d49091327442b70ddc)
- [Added examples for OpenAI, Bedrock, Anthropic, and VertexAI](https://github.com/SolaceDev/solace-ai-connector/commit/14d457db923c35bae049e26442d8266fa7153d13)
- [Updated old examples to latest version of their libraries and ensuring they work as expected](https://github.com/SolaceDev/solace-ai-connector/commit/645cd679a55cde1ee3bd895082b0c769e0f8f066)
- [Fixed bug with dynamic instantiation of vectorstore class](https://github.com/SolaceDev/solace-ai-connector/commit/16654940f284c3e6fbc80bf2196804a21f373880)

### CODE REVIEW

File: examples/anthropic_bedrock.yaml [hunk 1]
```diff
@@ -4,6 +4,22 @@
...
+# pip install langchain_aws
+# 
+# required ENV variables:
+# - SOLACE_BROKER_URL
+# - SOLACE_BROKER_USERNAME
+# - SOLACE_BROKER_PASSWORD
+# - SOLACE_BROKER_VPN
+# - AWS_BEDROCK_ANTHROPIC_CLAUDE_MODEL_ID
+
+
```
Comment (efunneko) [inline examples/anthropic_bedrock.yaml:22]: I like these additional comments!

File: examples/vector_store_search.yaml [hunk 2]
```diff
@@ -1,28 +1,40 @@
...
+  # eg: aws_secret_access_key: ${AWS_SECRET_ACCESS_KEY}
   - aws_credentials: &aws_credentials
       invoke:
         object:
           invoke:
             module: boto3
             function: Session
+            params:
+              keyword:
+                profile_name: default # The profile to choose from .aws/credentials
```
Comment (efunneko) [inline examples/vector_store_search.yaml:37]: I guess this was the secret sauce to get this working again

File: src/solace_ai_connector/common/utils.py [hunk 3]
```diff
@@ -118,6 +118,7 @@ def import_module(name, base_path=None, component_package=None):
                     ".components.general",
                     ".components.general.for_testing",
                     ".components.general.langchain",
+                    ".components.general.openai",
```
Comment (efunneko) [inline src/solace_ai_connector/common/utils.py:121]: Did it take you a long time to find this? Sorry - somehow it was missed in my commit for the openai component

**Review by efunneko — APPROVED** [review]
Looks good! Thanks for this cleanup - it has needed it for a while

**Comment by efunneko** [comment]
You can do the merge. Later I will kick off a release for it.

File: examples/vector_store_search.yaml (hunk 2 above)
Comment (cyrus2281) [reply examples/vector_store_search.yaml:37]: Actually, the main issues were from AWS configurations side

File: src/solace_ai_connector/common/utils.py (hunk 3 above)
Comment (cyrus2281) [reply src/solace_ai_connector/common/utils.py:121]: No, the error said it couldn't fine/load the module, so I just looked where langchain was added, and duplicated that process

**Pull request merged**